task7-2/expressionTree.c

```diff
@@ -0,0 +1,167 @@
+#include "expressionTree.h"
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+#define CAPACITY 256
+#define MEMORY_ALLOCATION_ERROR (-2)
+
+typedef struct Node {
+    char operation;
+    int number;
+    struct Node *left;
+    struct Node *right;
+} Node;
+
+typedef struct Tree {
+    Node *root;
+} Tree;
+
+Tree *initTree() {
+    Tree *tree = malloc(sizeof(Tree));
+    return tree;
+}
```

Comment:
```suggestion
Tree *initTree() {
    return сalloc(sizeof(Tree));
}
```
Иначе вызывающему возвращается структура с мусором в поле root. Который он, наверное, перезапишет, но а если нет?